src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -177,7 +177,19 @@ protected function compileJoinConstraint(array $clause)
     {
         $first = $this->wrap($clause['first']);
 
-        $second = $clause['where'] ? '?' : $this->wrap($clause['second']);
+        if ($clause['where']) {
+
```

Comment:
> Am I wrong?

Yes, you are, sorry. Our CS is zero blank lines following or just before a braces.